src/compute-categories.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Module that exports a function that takes a spec object that already has its
+ * `groups` property (see `fetch-groups.js`) and its `nightly.repository`
+ * property (see `compute-repository.js`) set as input, and that returns
+ * a list of categories for the spec.
+ *
+ * The function merely sets `browser` for now. Logic (and initial spec
+ * properties used to compute the list) will likely be adjusted over time.
+ */
+
+// Retrieve the list of groups and repositories that we know don't contain
+// specs targeted at browsers. That logic will also very likely evolve over
+// time, be it only to give the file a different name (the list of specs will
+// be expanded to contain specs in that "ignore" list)
+const { groups: nonbrowserGroups, repos: nonbrowserRepos } = require('./data/ignore.json');
+
+/**
+ * Exports main function that takes a spec object and returns a list of
+ * categories for the spec.
+ *
+ * Function may return an empty array. If the spec object contains a
+ * `categories` property, the list of categories is adjusted accordingly. For
+ * instance, if the spec object contains `+browser`, `browser` is added to the
+ * list. If it contains `-browser`, `browser` won't appear in the list. If it
+ * contains `reset`, the function does not attempt to compute a list but rather
+ * returns the list of categories in the spec object.
+ */
+module.exports = function (spec) {
+  if (!spec || !spec.groups) {
+    throw "Invalid spec object passed as parameter";
+  }
+
+  let list = [];
+  const requestedCategories = (typeof spec.categories === "string") ?
+    [spec.categories] :
+    (spec.categories || []);
+
+  // All specs target browsers by default unless the spec object says otherwise
+  if (!requestedCategories.includes("reset")) {
+    const browserGroup = !spec.groups.find(group => nonbrowserGroups[group.name]);
```

Comment:
if I read this correctly, having a non-browser group listed as co-owning a spec disqualifies the spec as a browser spec? if so, while that may be OK today, it feels like this should be documented as potentially wobbly

Reply:
I had "In general, I expect the logic in compute-categories.js to evolve based on non-browser specs that we add to the list" as a disclaimer in the commit message. I'll add something along these lines to the source code.

In particular, the logic is just meant to work today to tag the 4 specs in the list that don't target browsers, but I agree that we'll need to improve it. I would prefer to do that based on a concrete list of specs, meaning when we actually start adding non-browser specs.

(Typically, I cannot think of an example of a joint deliverable that would qualify as a browser spec but for which one of the groups would fall into the list of non browser groups. Do you have one in mind?)

Reply:
Well, I note the code already has a "Logic (and initial spec properties used to compute the list) will likely be adjusted over time" comment near the top. I'll add a more specific comment to that part.